graphql_compiler/query_formatting/common.py

```diff
@@ -34,91 +35,84 @@
 ######
 
 
-def _raise_invalid_type_error(name, expected_python_type_name, value):
+def _raise_invalid_type_error(
+    name: str, expected_python_types: Collection[Type], value: Any
+) -> NoReturn:
     """Raise a GraphQLInvalidArgumentError that states that the argument type is invalid."""
     raise GraphQLInvalidArgumentError(
-        "Invalid type for argument {}. Expected {}. Got value {} of "
-        "type {}.".format(name, expected_python_type_name, value, type(value).__name__)
+        f"Invalid type for argument {name}. Expected one of {expected_python_types}. Got value "
+        f"{value} of type {type(value).__name__} instead."
     )
 
 
-def _deserialize_anonymous_json_argument(expected_type: GraphQLScalarType, value: Any) -> Any:
-    """Deserialize argument. See docstring of deserialize_json_argument.
+_CUSTOM_SCALAR_DESERIALIZATION_FUNCTIONS = MappingProxyType(
+    {GraphQLInt.name: int, GraphQLFloat.name: float}
+)
 
-    Args:
-        expected_type: GraphQL type we expect.
-        value: object that can be interpreted as being of that type
 
-    Returns:
-        a value of the type produced by the parser of the expected type:
-            GraphQLDate: datetime.date
-            GraphQLDateTime: datetime.datetime with tzinfo=pytz.utc
-            GraphQLFloat: float
-            GraphQLDecimal: decimal.Decimal
-            GraphQLInt: int
-            GraphQLString: str
-            GraphQLBoolean: bool
-            GraphQLID: str
-
-    Raises:
-        ValueError if the value is not appropriate for the type. ValueError is chosen because
-        it is already the base case of exceptions raised by the GraphQL parsers.
-    """
-    allowed_types_for_graphql_type = {
-        GraphQLDate.name: (str,),
-        GraphQLDateTime.name: (str,),
-        GraphQLFloat.name: (str, float, int),
-        GraphQLDecimal.name: (str, float, int),
-        GraphQLInt.name: (int, str),
-        GraphQLString.name: (str,),
-        GraphQLBoolean.name: (bool,),
-        GraphQLID.name: (int, str,),
-    }
+def _get_json_scalar_deserialization_function(
+    graphql_type: GraphQLScalarType, expected_python_types: Tuple[Type, ...],
+) -> Callable[[str, Any], Any]:
+    """Return a function that deserializes a json serialized argument of the given GraphQL type."""
 
-    # Check for long integers, bypassing the GraphQLInt parser
-    if is_same_type(GraphQLInt, expected_type):
-        if isinstance(value, int):
-            return value
-        elif isinstance(value, str):
-            return int(value)
-        else:
-            raise ValueError(
-                "Unexpected type {}. Expected one of {}.".format(type(value), (int, str))
+    def _json_scalar_deserialization_function(name: str, value: Any) -> Any:
```

Comment:
Is there any benefit to this being defined as a function-inside-a-function, instead of being a proper top-level function that takes all the args it needs, and then being used by using `partial` to provide the fixed arguments such as `graphql_type` and `expected_python_types`?